provider-service/kfp/internal/suite_decoupled_test.go

```diff
@@ -0,0 +1,320 @@
+//go:build decoupled
+
+package internal
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/argoproj/argo-workflows/v3/pkg/apis/workflow"
+	"github.com/sky-uk/kfp-operator/provider-service/base/pkg/streams"
+	"io"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"net/http"
+	"path/filepath"
+	"testing"
+	"time"
+
+	argo "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+	"github.com/go-resty/resty/v2"
+	"github.com/jarcoal/httpmock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/sky-uk/kfp-operator/argo/common"
+	"github.com/sky-uk/kfp-operator/provider-service/base/pkg"
+	"github.com/sky-uk/kfp-operator/provider-service/base/pkg/streams/sinks"
+	"github.com/sky-uk/kfp-operator/provider-service/base/pkg/streams/sources"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var (
+	mockMetadataStore MockMetadataStore
+	mockKfpApi        MockKfpApi
+	k8sClient         dynamic.Interface
+	eventSource       *sources.WorkflowSource
+	webhookSink       *sinks.WebhookSink
+	eventFlow         streams.Flow[pkg.StreamMessage[*unstructured.Unstructured], pkg.StreamMessage[*common.RunCompletionEventData], error]
+	eventData         common.RunCompletionEventData
+	numberOfEvents    int
+)
+
+var (
+	argoWorkflowsGvr = schema.GroupVersionResource{
+		Group:    workflow.Group,
+		Version:  workflow.Version,
+		Resource: workflow.WorkflowPlural,
+	}
+)
+
+const (
+	defaultNamespace = "default"
+	providerName     = "kfp"
+	webhookUrl       = "/operator-webhook"
+)
+
+func TestModelUpdateEventSourceDecoupledSuite(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Run Completion EventSource Decoupled Suite")
+}
+
+func deleteAllWorkflows(ctx context.Context) error {
```

Comment:
we should move these helper functions below the test suite to reduce the noise

Reply:
yes we should changed in 847762891ac9e4c53e697be2fbd16d03bfb70260